src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+
+* `tosearch {Object|Array}` The structure to be searched for the supplied value
+* `value {Any}` The value to be found. This will be compared against the structure's members using `===` equality.
+* Returns: `{String|Integer|Undefined}` The first key whose value matches the one supplied, or `undefined` is no value matches
+
+### fluid.stableSort(array, func) 
+
+Applies a stable sorting algorithm to the supplied array and comparator (note that [Array.sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) in JavaScript is not specified
+to be stable). The algorithm used will be an insertion sort, which whilst quadratic in time, will perform well
+on small array sizes.
+* `array {Array of Any-Element}` The array to be sorted. This input array will be modified in place.
+* `func {Function: (a {Any-Element}, b {Any-Element}) → Number}` A comparator returning >0, 0, or <0 on pairs of elements representing their sort order (same contract as [Array.sort comparator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort))
+
+
+### fluid.compareStringLength(ascending)
+
+Produces a comparator function suitable for use with [Array.sort](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort) to sort an array of strings by length
+
+* `ascending {Booleanish}` `true` if a comparator is to be returned which sorts strings in descending order of length
+* Returns: `{Function: (a {String}, b {String}) → Number}` The comparator function to be used for sorting
+
+
+## Parsing and rendering
+
+### fluid.parseInteger(string)
+
+Returns the converted integer if the input string can be converted to an integer. Otherwise, returns `NaN`.
+
+* `string {String}` A string to be converted to an integer
+* Returns: `{Integer|NaN}` The integer value, or `NaN` if the string does not represent an integer 
+
+### fluid.stringTemplate(template, terms) {
+
+A simple string interpolation system. Accepts a template string with embedded tokens of the form `%value`. Returns a new string with the tokens replaced by the specified values.
+Keys and values can be of any data type that can be coerced into a string. Tokens are not delimited in the template with any syntax - the implementation will replace the longest token which
+can be sourced from `terms`.
+
+* `template {String}` A string with embedded tokens of the form `%value`
+* `terms {Object: String → Stringable}` A hash which maps tokens to their required interpolated values   
+
+
+## Handling models and paths
+
+### fluid.get(model, path)
+
+Accesses an element nested within a [model object](FrameworkConcepts.md#model-object] at the supplied (EL) path. `fluid.get` can
+operate various special behaviour in encountering particular kinds of material - for example, any object implementing a function named `resolvePathSegment` will
```

Comment:
This explanation of 'special behaviours' is not terribly clear; for example: "any object implementing a function named resolvePathSegment" - what does this mean? is this 'resolvePathSegment' function something the reader would create? what does it mean for an object to implement the function? ...